src/operations/map_reduce.ts

```diff
@@ -96,55 +103,52 @@ export class MapReduceOperation extends CommandOperation<MapReduceOptions, Docum
     const coll = this.collection;
     const map = this.map;
     const reduce = this.reduce;
-    let options = this.options;
 
     const mapCommandHash: Document = {
       mapReduce: coll.collectionName,
       map: map,
       reduce: reduce
     };
 
-    if (options.scope) {
-      mapCommandHash.scope = processScope(options.scope);
+    if (this.scope) {
+      mapCommandHash.scope = processScope(this.scope);
     }
 
     // Add any other options passed in
-    for (const n in options) {
-      // Only include if not in exclusion list
-      if (exclusionList.indexOf(n) === -1) {
-        mapCommandHash[n] = (options as any)[n];
+    for (const n in this) {
+      // Only include if defined on this
+      if (optionsAllowList.has(n)) {
```

Comment:
The other ops that do this check for null values / functions wondering if there's a way we can gain consistency here?

Reply:
Is there more ops than mapReduce that do this? I think MapReduce does this since the operation takes in functions it makes sense to double check the options don't have any (for example, `validator` sounds like a function but is a document). I can get more context on this if there's more